src/dataconnect/schemaMigration.ts

```diff
@@ -52,11 +66,35 @@ export async function diffSchema(schema: Schema): Promise<Diff[]> {
       displayInvalidConnectors(invalidConnectors);
     }
     if (incompatible) {
-      displaySchemaChanges(incompatible);
-      return incompatible.diffs;
+      displaySchemaChanges(incompatible, valMode);
+      diffs = incompatible.diffs;
     }
   }
-  return [];
+
+  if (experiments.isEnabled("fdccompatiblemode")) {
+    // If the validation mode is unset, then we also surface any additional optional STRICT diffs.
+    if (!schemaValidation) {
+      valMode = "STRICT";
+      setSchemaValidationMode(schema, valMode);
+      try {
+        await upsertSchema(schema, /** validateOnly=*/ true);
+        logLabeledSuccess("dataconnect", `Database schema is up to date.`);
+      } catch (err: any) {
+        if (err.status !== 400) {
```

Comment:
```suggestion
        if (err?.status !== 400) {
```
I think this will silence some of these warnings.